jolt-core/src/jolt/instruction/lte.rs

```diff
@@ -0,0 +1,134 @@
+
+        // Accumulator for LTU(x_{<s}, y_{<s})
+        let mut ltu_sum = lt_abs[0];
+        // Accumulator for EQ(x_{<s}, y_{<s})
+        let mut eq_prod = eq_abs[0];
+
+        for (ltu_i, eq_i) in ltu.iter().zip(&eq[1..]) {
+            ltu_sum += *ltu_i * eq_prod;
+            eq_prod *= eq_i;
+        }
+
+        // x_s * (1 - y_s) + EQ(x_s, y_s) * LTU(x_{<s}, y_{<s})
+        let lt = gt_msb[0] + eq_msb[0] * ltu_sum;
+
+        // EQ(x,y)
+        let eq = eq.iter().product::<F>();
+
+        // LTS(x,y) || EQ(x,y)
+        lt + eq - lt * eq
+    }
+
+    fn g_poly_degree(&self, C: usize) -> usize {
+        C + 1
+    }
+
+    fn subtables<F: JoltField>(
+        &self,
+        C: usize,
+        _: usize,
+    ) -> Vec<(Box<dyn LassoSubtable<F>>, SubtableIndices)> {
+        vec![
+            (Box::new(GtMSBSubtable::new()), SubtableIndices::from(0)),
+            (Box::new(EqMSBSubtable::new()), SubtableIndices::from(0)),
+            (Box::new(LtuSubtable::new()), SubtableIndices::from(1..C)),
+            (Box::new(EqSubtable::new()), SubtableIndices::from(0..C)),
+            (Box::new(LtAbsSubtable::new()), SubtableIndices::from(0)),
+            (Box::new(EqAbsSubtable::new()), SubtableIndices::from(0)),
+        ]
+    }
+
+    fn to_indices(&self, C: usize, log_M: usize) -> Vec<usize> {
+        chunk_and_concatenate_operands(self.0, self.1, C, log_M)
+    }
+
+    fn lookup_entry(&self) -> u64 {
+        ((self.0 as i32) <= (self.1 as i32)).into()
+    }
+
+    fn random(&self, rng: &mut StdRng) -> Self {
+        Self(rng.next_u32() as u64, rng.next_u32() as u64)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use ark_bn254::Fr;
+    use ark_std::test_rng;
+    use rand_chacha::rand_core::RngCore;
+
+    use crate::{jolt::instruction::JoltInstruction, jolt_instruction_test};
+
+    use super::LTEInstruction;
+
+    #[test]
+    fn lte_instruction_e2e() {
+        let mut rng = test_rng();
+        const C: usize = 4;
+        const M: usize = 1 << 16;
+
+        // Random
+        for _ in 0..256 {
+            let x = rng.next_u32();
+            let y = rng.next_u32();
+
+            let instruction = LTEInstruction(x as u64, y as u64);
+
+            jolt_instruction_test!(instruction);
+        }
+
+        // Ones
+        for _ in 0..256 {
+            let x = rng.next_u32();
+            jolt_instruction_test!(LTEInstruction(x as u64, x as u64));
+        }
+
+        // Edge-cases
+        let u32_max: u64 = u32::MAX as u64;
+        let instructions = vec![
+            LTEInstruction(100, 0),
+            LTEInstruction(0, 100),
+            LTEInstruction(1, 0),
+            LTEInstruction(0, u32_max),
+            LTEInstruction(u32_max, 0),
+            LTEInstruction(u32_max, u32_max),
+            LTEInstruction(u32_max, 1 << 8),
+            LTEInstruction(1 << 8, u32_max),
+        ];
+        for instruction in instructions {
+            jolt_instruction_test!(instruction);
+        }
+    }
```

Comment:
We should ideally break this into seperate tests for u32 and u64 respectively to keep it consistent